### ISSUE TITLE
OSS and security

### ISSUE THREAD

**Issue by KKIMONK** [opensemanticsearch/open-semantic-search-doc#7]
Hello,
I have a question about security management with OSS. How is it managed? I take an example, I'm using OSS like enterprise search engine on a server and from an user workstations under Windows, the requests are made either under a browser or under the windows explorer. 
Are rights and windows ACLs inherited by OSS ?
In other words, when I make a query search, does the result provided by OSS respect the security set up (rights and ACLs?)
Help would be appreciated?
Kader

**Comment by mosea3** [comment]
Hello Kader,

As of my knowledge:
- OSS indexes with a generic user (or any other) everything accessible under its account
- all of this is retrieved in a search query
- all of this is also previewable (you can turn it off in a OSS-setting)
- built-in OSS: anythin is showed with preview accessible to the generic index user

So: all users see what the generic indexing user has access to..

On my fork, there's a free preview as well, but the respective file is just referenced (UNC Path), not served - thus:
If you don't have access to that file under your own Identity you won't get it.

I was either way checking how to map the ACLs of windows (via automatic Kerberos Authentication towards the Unix Server) to OSS in order to limit preview to actual ACLs.

If you got a big enough system environment to set up a demo site (1 unix server for OSS, 1 Windows file server or NAS), we could team up and realize this together.

Multitple companies were already interested in using OSS as corporate Search Engine, but lack of ACLs they didn't implement it.
This Feature could attract more entities to OSS - I guess.

best regards


Andy


Am 23.10.2019 um 10:18 schrieb KKIMONK <notifications@github.com<mailto:notifications@github.com>>:


Hello,
I have a question about security management with OSS. How is it managed? I take an example, I'm using OSS like enterprise search engine on a server and from an user workstations under Windows, the requests are made either under a browser or under the windows explorer.
Are rights and windows ACLs inherited by OSS ?
In other words, when I make a query search, does the result provided by OSS respect the security set up (rights and ACLs?)
Help would be appreciated?
Kader
